File: Algorithms/Sorting/Counting_Sort_2/main.py
### Counting Sort 2 - Solution

from collections import Counter

def countingSort(arr):
    cnt = Counter(arr)
    idxCounter = tuple(cnt[i] for i in range(100))
    for j in range(100):
        if idxCounter[j] != 0:
            for _ in range(idxCounter[j]):
                print(j, end=' ')

n = int(input())
arr = tuple(map(int, input().split()[:n]))
countingSort(arr)

File: Algorithms/Sorting/Counting_Sort_2/main.cpp
// Counting Sort 2 - Solution

#include <iostream>
#include <cstring>

void countingSort(int n, int arr[]) {
    int zeroArr[100];
    std::memset(zeroArr, 0, sizeof zeroArr);
    for (int i{}; i < n; i++)
        zeroArr[arr[i]]++;
    for (int j{}; j < 100; j++) {
        if (zeroArr[j] != 0) 
            for (int k{}; k < zeroArr[j]; k++)
                std::cout << j << " ";
    }
}

int main() {
    int n;
    std::cin >> n;
    int arr[n];
    for (int i{}; i < n; i++)
        std::cin >> arr[i];
    countingSort(n, arr);

    return 0;
}